File: 1175 - Prime Arrangements.py
import math


class Solution:
    def numPrimeArrangements(self, n: int) -> int:
        if n == 1:
            return 1
        primeCount = 0

        def isPrime(i):
            for j in range(2, int(math.sqrt(i) + 1)):
                if i % j == 0:
                    return False
            return True
        for i in range(2, n + 1):
            if isPrime(i):
                primeCount += 1
        m = 10 ** 9 + 7
        res = (math.factorial(primeCount) * math.factorial(n - primeCount)) % m
        return int(res)

File: 1175 - Prime Arrangements.cs
public class Solution
{
    private const int m = 1000000000 + 7;
    public int NumPrimeArrangements(int n)
    {
        if (n == 1) return 1;
        int primeCount = 0;
        for (int i = 2; i <= n; i++)
        {
            if (isPrime(i))
                primeCount++;
        }

        long res = (fac(primeCount) * fac(n - primeCount));
        return (int)(res % m);
    }

    private long fac(int n)
    {
        long res = 1;
        for (int i = 2; i <= n; i++)
        {
            res *= i;
            res %= m;
        }
        return res;
    }
    private bool isPrime(int n)
    {
        for (int i = 2; i * i <= n; i++)
        {
            if (n % i == 0)
                return false;
        }
        return true;
    }
}